### PR TITLE
[Dev/51] Problem List page

### PR DESCRIPTION
**Pull request by nyngwang**
image

### CODE REVIEW

File: src/app/problem/Modal/CreateProblemModal.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+export const useProblemList = () => {
+    const [tags, setTags] = useState(undefined);
+    const addTag = (tag) => {
+        tags.push(tag);
+        setTags(tags);
+    };
+
+    return {tags, setTags, addTag}
+};
```
Comment (Johnny850807) [inline src/app/problem/Modal/CreateProblemModal.js:16]: don't write unnecessary code by copy-and-paste

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

File: src/app/problem/Modal/CreateProblemModal.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    useEffect(() => {
+
+    })
+
+    const handleSubmit = e => {
+        e.preventDefault();
+        const form = formRef.current;
+
+        if (form.checkValidity()) {
+            problemService.createProblem(problemName, tags)
```
Comment (Johnny850807) [inline src/app/problem/Modal/CreateProblemModal.js:32]: I don't think the `tags` is the parameter

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review

File: src/app/problem/Modal/CreateProblemModal.js [hunk 3]
```diff
@@ -29,8 +19,9 @@ const CreateProblemModal = ({ show, onClose, onProblemCreated }) => {
         const form = formRef.current;
 
         if (form.checkValidity()) {
-            problemService.createProblem(problemName, tags)
+            problemService.createProblem(problemName)
                 .then(problem => {
+                    console.log("DEBUG----", problem)
```
Comment (Johnny850807) [inline src/app/problem/Modal/CreateProblemModal.js:24]: remove

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Last but not least

**Review by Johnny850807 — APPROVED** [review]
LGTM

File: src/app/problem/Modal/CreateProblemModal.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import {ModalHeader} from "../../commons/modals/ModalHeader";
+
+
+const CreateProblemModal = ({ show, onClose, onProblemCreated }) => {
+    const [problemName, setName] = useState('');
+    const closeIconRef = createRef(), formRef = createRef(), nameInputRef = createRef()
+
+    useEffect(() => {
+
+    })
```
Comment (Chaoyuuu) [inline src/app/problem/Modal/CreateProblemModal.js:15]: Remove it !

**Review by Chaoyuuu — APPROVED** [review]
One thing to fix. LGTM.

**Pull request merged**